x/distribution/keeper/allocation.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// Allocate fees handles distribution of the collected fees
+func (k Keeper) AllocateFees(ctx sdk.Context) {
+	ctx.Logger().With("module", "x/distribution").Error(fmt.Sprintf("allocation height: %v", ctx.BlockHeight()))
+
+	// if there is no power in the system nothing should be allocated
+	bondedTokens := k.stakeKeeper.TotalPower(ctx).TruncateInt()
```

Comment:
When do we actually expect this to happen? I think maybe we should panic here - how would we ever produce blocks with zero power?

Reply:
Haha sorry I thought I had asked this before but didn't find the comment - OK, makes sense!